src/routes/api/sections/bookSection.ts

```diff
@@ -0,0 +1,176 @@
+import { Request, Response } from 'express';
+import timeBlockModel from '../../../models/timeBlock.model';
+import sectionModel from '../../../models/section.model';
+import Room from '../../../models/room.model';
+import { DateTime } from 'luxon';
+
+const bookSectionRoute = async (req: Request, res: Response): Promise<void> => {
+    const id = req.params.id;
+    const { emails, startsAt, endsAt } = req.body;
+
+    // Checks for needed request body variables
+    if (!id || !emails || !startsAt || !endsAt) {
+        res.status(400);
+        return;
+    }
+
+    // Retrieves section with the section id given
+    const sectionInformation = await sectionModel.findOne({ _id: id }).catch((error) => {
+        res.status(500).json({ error });
+        return;
+    });
+
+    if (!sectionInformation) {
+        res.status(404);
+        return;
+    }
+
+    // Retrieves room that the section corresponds to
+    const room = await Room.findOne({ _id: sectionInformation.roomId }).catch((error) => {
+        res.status(404).json({ error });
+        return;
+    });
+
+    if (!room) {
+        res.status(404);
+        return;
+    }
+
+    const start = DateTime.fromISO(String(startsAt), { zone: 'America/Toronto' });
+    const end = DateTime.fromISO(String(endsAt), { zone: 'America/Toronto' });
+
```

Comment:
You should add a check to see if the `DateTime`s are valid using the `.invalid` property before continuing